subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -544,6 +549,12 @@ public AbstractTestTask setTestNameIncludePatterns(List<String> testNamePattern)
         return this;
     }
 
+    @Option(option = "fail-fast", description = "Stops test execution after the first failed test")
+    public AbstractTestTask setFailFast(boolean failFast) {
```

Comment:
This basically already exposed the functionality for the task type. In the build script it could look as follows.

```
test {
    failFast = true
}
```

That's what I meant by "expose a property on the task type".

Reply:
When using @option to accept values from user for a plugin, currently if a boolean variable is declared as false plugin, it can be enabled by passing --variable. However, since the variable needs to be true in most case and needs to be disabled on demand. How can this operation be performed from a consuming project?